FxSsh/Services/UserauthService.cs

```diff
@@ -35,33 +36,76 @@ private void HandleMessage(RequestMessage message)
             switch (message.MethodName)
             {
                 case "publickey":
-                    var msg = Message.LoadFrom<PublicKeyRequestMessage>(message);
-                    HandleMessage(msg);
+                    var keyMsg = Message.LoadFrom<PublicKeyRequestMessage>(message);
+                    HandleMessage(keyMsg);       
                     break;
                 case "password":
+                    var msg = Message.LoadFrom<PasswordRequestMessage>(message);
+                    HandleUserLogonMessage(msg);
+                    break;
                 case "hostbased":
                 case "none":
+                    _session.SendMessage(new FailureMessage());
+                    break;
                 default:
                     _session.SendMessage(new FailureMessage());
                     break;
             }
         }
 
+        private void HandleUserLogonMessage(PasswordRequestMessage message)
+        {
+            var verifed = false;
+
+            var args = new UserauthArgs(base._session, message.Username, message.Password);
+            if (Userauth != null)
+            {
+                Userauth(this, args);
+                verifed = args.Result;
+            }
+
+            if (verifed)
+            {
+                _session.RegisterService(message.ServiceName, args);
+
+                if (Succeed != null)
+                    Succeed(this, message.ServiceName);
```

Comment:
I'd like use new grammar
```suggestion
                Succeed.?(this, message.ServiceName);
```